extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -455,6 +472,9 @@ export class AmpAd3PImpl extends AMP.BaseElement {
 
   /** @override  */
   unlayoutCallback() {
+    this.unlisteners_.forEach((unlisten) => unlisten());
+    this.unlisteners_.length = 0;
```

Comment:
should you set it back to empty array instead of overwriting length? can you even overwrite length?

Reply:
resolved offline. it's an officially supported way.